File: 11week/2. Maximum Points You Can Obtain from Cards.cpp
class Solution {
public:
    // 생각안하고 풀면 dp로 잘못 풀 수 있음
    
    int maxScore(vector<int>& cardPoints, int k) {
        int race = 0;
        for(int i = 0; i < k; i++){
            race += cardPoints[i];
        }
        
        int left = k - 1;
        int right = cardPoints.size() - 1;
        int ans = race;
        
        while(left >= 0){
            race -= cardPoints[left--];
            race += cardPoints[right--];
            ans = max(ans, race);
        }
        return ans;
    }
};

File: 11week/홍준/2.maximum-points-you-can-obtain-from-cards.py
## https://leetcode.com/problems/maximum-points-you-can-obtain-from-cards/

from typing import List

class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        size = len(cardPoints)
        answer = sum(cardPoints[size-k:size])
        partial_sum = answer
        for i in range(k):
            partial_sum += cardPoints[(size + i) % size] - cardPoints[(size - k + i) % size]
            answer = max(answer, partial_sum)
        return answer